tools/tertiary-analysis/decoupler/decoupler_pathway_inference.py

```diff
@@ -93,8 +119,15 @@
 if args.output is None:
     raise ValueError("Please specify either -o or --output")
 
-# read in the AnnData input file
-adata = ad.read_h5ad(args.input_anndata)
+# detect based on input file extension if the input file is AnnData or matrix
+if args.input.endswith(".h5ad"):
```

Comment:
how about input file with `.h5` extension

Reply:
It is not needed, since this is internally controlled with the galaxy wrapper, which will transform the filename forcefully into input.h5ad . Also the extension .h5 is generally used for any HDF5 file, not just AnnData, the community tends to favour .h5ad for anndata (although it is just a convention).